src/Services/RegistrationsInfoCacheService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Core.Services;
+using Lykke.Service.Registration;
+using MessagePack;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Services
+{
+    public class RegistrationsInfoCacheService : IRegistrationsInfoCacheService
+    {
+        private readonly IDistributedCache _cache;
+        private readonly ILykkeRegistrationClient _registrationClient;
+        private readonly DistributedCacheEntryOptions _cacheOptions;
+        private const string RegistrationsInfoPrefix = "RegistrationsInfo";
+        private readonly SemaphoreSlim _lockSlim = new SemaphoreSlim(1, 1);
+
+        public RegistrationsInfoCacheService(
+            IDistributedCache cache,
+            ILykkeRegistrationClient registrationClient,
+            TimeSpan cacheExpirationPeriod
+            )
+        {
+            _cache = cache;
+            _registrationClient = registrationClient;
+            _cacheOptions = new DistributedCacheEntryOptions {AbsoluteExpirationRelativeToNow = cacheExpirationPeriod};
+        }
+        
+        public async Task<long> GetRegistrationsCountAsync()
+        {
+            await _lockSlim.WaitAsync();
```

Comment:
There is already extension method for ```IDistributedCache``` that takes care about multithreading - https://github.com/LykkeCity/LykkePublicAPI/blob/master/src/Services/CacheExtensions.cs#L29 will be better to use it